File: app/src/main/java/com/androidevlinux/percy/UTXO/data/models/coinmarketcap/CoinMarketCapChartData.java
package com.androidevlinux.percy.UTXO.data.models.coinmarketcap;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

/**
 * Created by Ryan on 2/3/2018.
 */

public class CoinMarketCapChartData implements Serializable
{

    @SerializedName("market_cap_by_available_supply")
    @Expose
    private List<List<Float>> marketCapByAvailableSupply = null;
    @SerializedName("price_btc")
    @Expose
    private List<List<Float>> priceBtc = null;
    @SerializedName("price_usd")
    @Expose
    private List<List<Float>> priceUsd = null;
    @SerializedName("volume_usd")
    @Expose
    private List<List<Float>> volumeUsd = null;
    private final static long serialVersionUID = 4430369643668729100L;

    public List<List<Float>> getMarketCapByAvailableSupply() {
        return marketCapByAvailableSupply;
    }

    public void setMarketCapByAvailableSupply(List<List<Float>> marketCapByAvailableSupply) {
        this.marketCapByAvailableSupply = marketCapByAvailableSupply;
    }

    public CoinMarketCapChartData withMarketCapByAvailableSupply(List<List<Float>> marketCapByAvailableSupply) {
        this.marketCapByAvailableSupply = marketCapByAvailableSupply;
        return this;
    }

    public List<List<Float>> getPriceBtc() {
        return priceBtc;
    }

    public void setPriceBtc(List<List<Float>> priceBtc) {
        this.priceBtc = priceBtc;
    }

    public CoinMarketCapChartData withPriceBtc(List<List<Float>> priceBtc) {
        this.priceBtc = priceBtc;
        return this;
    }

    public List<List<Float>> getPriceUsd() {
        return priceUsd;
    }

    public void setPriceUsd(List<List<Float>> priceUsd) {
        this.priceUsd = priceUsd;
    }

    public CoinMarketCapChartData withPriceUsd(List<List<Float>> priceUsd) {
        this.priceUsd = priceUsd;
        return this;
    }

    public List<List<Float>> getVolumeUsd() {
        return volumeUsd;
    }

    public void setVolumeUsd(List<List<Float>> volumeUsd) {
        this.volumeUsd = volumeUsd;
    }

    public CoinMarketCapChartData withVolumeUsd(List<List<Float>> volumeUsd) {
        this.volumeUsd = volumeUsd;
        return this;
    }

}

File: app/src/main/java/com/androidevlinux/percy/UTXO/data/models/coinmarketcap/CoinMarketCapChartData.kt
package com.androidevlinux.percy.UTXO.data.models.coinmarketcap

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

import java.io.Serializable

class CoinMarketCapChartData : Serializable {

    @SerializedName("market_cap_by_available_supply")
    @Expose
    var marketCapByAvailableSupply: List<List<Float>>? = null
    @SerializedName("price_btc")
    @Expose
    var priceBtc: List<List<Float>>? = null
    @SerializedName("price_usd")
    @Expose
    var priceUsd: List<List<Float>>? = null
    @SerializedName("volume_usd")
    @Expose
    var volumeUsd: List<List<Float>>? = null

    fun withMarketCapByAvailableSupply(marketCapByAvailableSupply: List<List<Float>>): CoinMarketCapChartData {
        this.marketCapByAvailableSupply = marketCapByAvailableSupply
        return this
    }

    fun withPriceBtc(priceBtc: List<List<Float>>): CoinMarketCapChartData {
        this.priceBtc = priceBtc
        return this
    }

    fun withPriceUsd(priceUsd: List<List<Float>>): CoinMarketCapChartData {
        this.priceUsd = priceUsd
        return this
    }

    fun withVolumeUsd(volumeUsd: List<List<Float>>): CoinMarketCapChartData {
        this.volumeUsd = volumeUsd
        return this
    }

    companion object {
        private const val serialVersionUID = 4430369643668729100L
    }

}

